pkg/cmd/admin/node/node.go

```diff
@@ -0,0 +1,117 @@
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const ManageNodeCommandName = "manage-node"
+
+const manageNodeLongDesc = `Manage node operations: schedulable, evacuate and list-pods.
+
+Examples:
+
+	# Block accepting any new pods on nodes
+	$ %[1]s <mynode> --schedulable=false
+
+	# Mark selected nodes as schedulable
+	$ %[1]s --selector="<env=dev>" --schedulable=true
+
+	# Migrate selected pods
+	$ %[1]s <mynode> --evacuate --pod-selector="<service=myapp>"
+
+	# Show pods that will be migrated
+	$ %[1]s <mynode> --evacuate --dry-run --pod-selector="<service=myapp>"
+
+	# List all pods on nodes
+	$ %[1]s <mynode1> <mynode2> --list-pods
+`
+
+func NewCommandManageNode(f *clientcmd.Factory, commandName, fullName string, out io.Writer) *cobra.Command {
+	var schedulable, evacuate, listpods bool
+
+	opts := &NodeOptions{}
+	schedulableOp := &SchedulableOptions{Options: opts}
+	evacuateOp := &EvacuateOptions{Options: opts}
+	listpodsOp := &ListPodsOptions{Options: opts}
+
+	cmd := &cobra.Command{
+		Use:   commandName,
+		Short: "Manage node operations: schedulable, evacuate, list-pods",
+		Long:  fmt.Sprintf(manageNodeLongDesc, fullName),
+		Run: func(c *cobra.Command, args []string) {
+
+			if err := ValidOperation(c); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := opts.Complete(f, c, args, out); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+
+			if err := opts.Validate(); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			// Cross op validations
+			if evacuateOp.DryRun && !evacuate {
+				err := errors.New("--dry-run is only applicable for --evacuate")
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			var err error
+			if c.Flag("schedulable").Changed {
+				schedulableOp.Schedulable = schedulable
+				err = schedulableOp.Run()
+			} else if evacuate {
+				err = evacuateOp.Run()
+			} else if listpods {
+				err = listpodsOp.Run()
+			}
+			kcmdutil.CheckErr(err)
+		},
+	}
+	flags := cmd.Flags()
+
+	// Supported operations
+	flags.BoolVar(&schedulable, "schedulable", false, "Control pod schedulability on the node.")
+	flags.BoolVar(&evacuate, "evacuate", false, "Migrate all/selected pods on the node.")
+	flags.BoolVar(&listpods, "list-pods", false, "List all/selected pods on the node. Printer flags --output, etc. are only valid for this option.")
+
+	// Common optional params
+	flags.StringVar(&opts.PodSelector, "pod-selector", "", "Label selector to filter pods on the node. Optional param for --evacuate or --list-pods")
+	flags.StringVar(&opts.Selector, "selector", "", "Label selector to filter nodes. Either pass one/more nodes as arguments or use this node selector")
+
+	// Operation specific params
+	evacuateOp.AddFlags(cmd)
+	listpodsOp.AddFlags(cmd)
+
+	return cmd
+}
+
+func ValidOperation(c *cobra.Command) error {
+	numOps := 0
+	if c.Flag("schedulable").Changed {
+		numOps++
+	}
+	if c.Flag("evacuate").Changed {
```

Comment:
I think that only `schedulable` is based on the `Changed` check.  The rest ought to be based on trueness.  `manage-nodes --schedulable=false --evacuate=false` should be valid, right?